src/applications/gi/components/vet-tec/EybVetTecCalculatorForm.jsx

```diff
@@ -0,0 +1,191 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import recordEvent from 'platform/monitoring/record-event';
+import { formatCurrency } from '../../utils/helpers';
+import { ariaLabels } from '../../constants';
+import Dropdown from '../Dropdown';
+import { calculatorInputChange } from '../../actions';
+import { connect } from 'react-redux';
+import RadioButtons from '../RadioButtons';
+
+class EybVetTecCalculatorForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const selectedProgram =
+      this.props.selectedProgram !== ''
+        ? this.props.selectedProgram
+        : this.props.preSelectedProgram;
+
+    this.state = {
+      tuitionFees: 0,
+      scholarships: 0,
+      programName: selectedProgram,
+    };
+  }
+
+  setProgramFields = programName => {
+    if (programName) {
+      const program = this.props.institution.programs.find(
+        p => p.description.toLowerCase() === programName.toLowerCase(),
+      );
+      if (program) {
+        const field = 'EYBVetTecProgram';
+        const value = {
+          vetTecTuitionFees: this.state.tuitionFees,
+          vetTecProgramName: this.state.programName,
+          vetTecScholarships: this.state.scholarships,
+          vetTecProgramFacilityCode: this.props.institution.facilityCode,
+        };
+        this.props.calculatorInputChange({ field, value });
+      }
+    }
+  };
+
+  handleApprovedProgramsChange = event => {
+    const vetTecProgramName = event.target.value;
+    const program = this.props.institution.programs.find(
+      p => p.description.toLowerCase() === vetTecProgramName.toLowerCase(),
+    );
```

Comment:
This is duplicate code of line 29 - 31. Would it be better to have a common function?